From b69008b5b114892bdc702e3a753b7a01bb4ca836 Mon Sep 17 00:00:00 2001
From: Fabrizio Castro <fabrizio.castro.jz@renesas.com>
Date: Mon, 15 Aug 2022 15:56:03 +0100
Subject: [PATCH 12/18] Bluetooth: hci_renesas: Complete the setup regardless
 of firmware load

The firmware loading step doesn't currently work, and the current
implementation of the driver leaves the serial port with the wrong
configuration, preventing the system from using the bluetooth device
further.
Disregard the outcome of the firmware loading step for now, so that we
can use the bluetooth device.

Signed-off-by: Fabrizio Castro <fabrizio.castro.jz@renesas.com>
---
 drivers/bluetooth/hci_renesas.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/drivers/bluetooth/hci_renesas.c b/drivers/bluetooth/hci_renesas.c
index 1f3000338..1b704a858 100644
--- a/drivers/bluetooth/hci_renesas.c
+++ b/drivers/bluetooth/hci_renesas.c
@@ -332,11 +332,13 @@ static int renesas_setup(struct hci_uart *hu)
 	hci_uart_set_flow_control(hu, true);
 
 	err = renesas_load_firmware(hu->hdev, FIRMWARE_DA14531);
-	if (err)
-		return err;
-
-	/* wait for HCI application to start */
-	usleep_range(8000, 10000);
+	if (err) {
+		bt_dev_warn(hu->hdev, "Continuing despite being unable to "
+			    "load the firmware");
+	} else {
+		/* wait for HCI application to start */
+		usleep_range(8000, 10000);
+	}
 
 	rdata->state = STATE_FW_BOOTED;
 
-- 
2.43.0

